workspace/viewer/PdfJsViewer.ts

```diff
@@ -478,29 +480,30 @@ export class PdfJsViewer {
       canvas.removeClass("shadow");
     }
     const canvasOffset = canvas.position();
-    const textLayer = $(".textLayer", this.jqRoot);
+    const textLayer = parent.find(".textLayer");
     textLayer.css({
       top : canvasOffset.top,
       left : canvasOffset.left
     });
   }
 
   resetCanvas() {
-    if (this.pdfPageView) {
-      this.pdfPageView.destroy();
-    }
-    this.pdfPageView = undefined;
+	  for (let page of this.loadedPages){
+		  page.destroy();
+	  }
+	  this.loadedPages = [];
     this.jqRoot.empty();
-    this.queue.clear();
+    this.queue.clear(); 
   }
 
-  private openCurrentPage() {
-    if (this.currentFileUrl) {
-      let lastCompletedMainFileId = (this.queue.isEmpty() && this.queue.lastCompleted)
-          ? this.queue.lastCompleted.mainFileId : undefined;
-      this.show(this.currentFileUrl, this.currentPage, true, lastCompletedMainFileId);
+  public showPage(page: number) {
+    if (page > this.numPages){
+      page = this.numPages;
     }
+    let pageWrapper = this.loadedPages.find(x => x.id === page).div;
```

Comment:
Thank you for the advice. 
Yes, it's defined by library. `div` is the variable of `PDFPageView` object. It stores `div`, which contains view of page.
`id` is also variable of this object like `div`.

Reply:
I haven't found any information about thin arrow functions neither in TypeScript nor in ES5/6. They exist only in CoffeeScript, but can be added in future.

Reply:
Oh yup, sorry. I thought they were in TS as well. We can replace them with just `function` but this will probably make this "lambda" way more wordy, so let's leave as is. 
However, you need to do something when `find` returns `undefined` which seems to be possible. For instance, `resetCanvas` call resets the array, but leaves `currentfile` as is, so subsequent call to `showPage` will crash.

Reply:
Yes, you are right. We associate our canvas with one pdf file, so resetting the canvas should also set `currentfile` to `undefined`.
Done.